### DISCUSSION TITLE
Is it possible for Instancio to produce two identical objects?

### DISCUSSION THREAD

**Discussion by blackdiz** [instancio/instancio#392] [Q&A] [answered]
Hi, Arman Sharif, thanks for the impressive library! I wonder if Instancio uses Random to generate data, is it possible to generate objects with the same values?

**Comment by armandino** [comment, marked as answer]
@blackdiz, thank you! Yes, it's possible by providing a seed value. Here's an example:

```java
long seed = 1234;
Person person1 = Instancio.of(Person.class).withSeed(seed).create();
Person person2 = Instancio.of(Person.class).withSeed(seed).create();

assertThat(person1).isEqualTo(person2);
```

Hope this helps!

**Reply by blackdiz** [reply]
Thank you for the quick reply!
Sorry for my misleading question. I actually want to know if it's possible to get identical objects when I use Intansio without providing a custom seed.
Here's my use case, I have a class (let's say `Person` with a field named `name`) and a method to update a `Person` object with a `PersonDto` object (assuming they have the same fields). I write a test like:
```
Person person = Instancio.of(Person.class).create();
PersonDto personDto = Instancio.of(PersonDto.class).create();
update(person, personDto);

assertThat(person).haveFieldWithValue("name", personDto.getName());
```
In the test, I want to test if the `update` correctly updates the `name` in `person` with the value of `name` in `personDto`, if `person.name`'s value is the same as `personDto.name` from the beginning, the test doesn't actually test the `update`.

**Reply by armandino** [reply]
@blackdiz no worries. Could you please clarify which objects are identical in the above example? Does the `update` method call `person.setName(dto.getName())`?

**Reply by blackdiz** [reply]
Yes, let's say the `update` is:
```
private void update(Person person, PersonDto personDto) {
    person.setName(personDto.getName());
}
```
If `Person person = Instancio.of(Person.class).create();` gives me a `person` named "John" and `PersonDto personDto = Instancio.of(PersonDto.class).create();` also gives me a `personDto` named "Json", then I cannot be sure that `assertThat(person).haveFieldWithValue("name", personDto.getName());` passed because the `update` method works or their `name` is the same from the very beginning.

**Reply by armandino** [reply]
I see what you mean. By default, generated strings are random lengths between 3 and 10 chars long. The probability of generated strings being equal is quite low. In addition, every time the test is run, new values are generated. I think your test is perfectly fine as it is.

However, if you are still concerned, you could increase the default string length range to 20-30 characters using [instancio.properties](https://www.instancio.org/user-guide/#overriding-settings-using-a-properties-file). Then the probability of generating identical strings would pretty close to zero, similar to a random UUID.

**Reply by blackdiz** [reply]
I see, thank you for your informative answers!

**Reply by armandino** [reply]
@blackdiz, any time! :)
